homeassistant/components/integration/sensor.py

```diff
@@ -254,6 +320,31 @@ def _unit(self, source_unit: str) -> str:
 
         return self._unit_template.format(integral_unit)
 
+    def _derive_and_set_attributes_from_state(self, source_state: State) -> None:
+        unit = source_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        if unit is not None:
+            self._unit_of_measurement = self._unit(unit)
```

Comment:
I realize this is old code, but I think we should motivate in a comment why we don't update the unit here
```suggestion
            self._unit_of_measurement = self._unit(unit)
```

Reply:
> we don't update the unit here

What do you mean with this? Do you mean the other unit-like attributes?

Reply:
I meant we should motivate why we don't update our unit of measurement if the source sensor's unit of measurement is `None`, which is a valid unit for unitless quantities:

```py
        unit = source_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
        if unit is not None:
            self._unit_of_measurement = self._unit(unit)
```

Is it maybe a bug, and the code should be changed like this:
```py
        if ATTR_UNIT_OF_MEASUREMENT in source_state.attributes:
            self._unit_of_measurement = source_state.attributes[ATTR_UNIT_OF_MEASUREMENT]
```

Reply:
`self._unit()` is required because the unit will never be the same as the source, self._unit derives the new unit

Reply:
Thanks for joining the discussion @dgomes ! That is helpful. 

I renamed the method to `_integrate_unit_over_time` and explained that deriving the unit for the integral only makes sense if the source unit is defined.

Reply:
I still don't see a reason we should not update this sensor's unit of measurement just because the source sensor's unit is None. I still think it's a bug, and the code should be changed like I suggested above.

Reply:
the generated unit is the source unit with a new suffix and prefix

it would be something like **k**None**h** - makes no sense to me...

Reply:
Yes, but my point is we should then set the unit to `None`, not leave it as it was before. The code now does this, the old code didn't.